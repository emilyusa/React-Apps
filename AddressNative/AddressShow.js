import React from 'react';
import {Button, Text, View} from 'react-native';
import styles from "./elf-styles";


export default class AddressShow extends React.Component {
    
    render() {
        return (
            <View className="App">
                <Text className="App-intro">firstName:{this.props.address.firstName}
                </Text>
                
                <Text className="App-intro">lastName:{this.props.address.lastName}
                </Text>
                
                <Text className="App-intro">Street:{this.props.address.street}
                </Text>
                
                <Text className="App-intro">City:{this.props.address.city}
                </Text>
                
                <Text className="App-intro">State:{this.props.address.state}
                </Text>
                
                <Text className="App-intro">Zip:{this.props.address.zip}
                </Text>

                <Text className="App-intro">Phone:{this.props.address.phone}
                </Text>

                <Text className="App-intro">Websit:{this.props.address.website}
                </Text>

                <Text className="App-intro">Email:{this.props.address.email}
                </Text>

                <Text className="App-intro">Contact:{this.props.address.contact}
                </Text>

                <Button
                    title="setAddressLast"
                    style={styles.button}
                    onPress={(event) => this.props.setAddress(-1, event)}>
                </Button>
                <Button
                    title="setAddressNext"
                    style={styles.button}
                    onPress={(event) => this.props.setAddress(1, event)}>
                </Button>

                              
          </View>
        );
    }
}

