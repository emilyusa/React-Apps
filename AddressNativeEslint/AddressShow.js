import React from 'react';
import { Button, Text, View } from 'react-native';
import styles from './elf-styles';
import PropTypes from 'prop-types';

export default class AddressShow extends React.Component {
    render() {
        return (
            <View style={styles.displayArea}>
                <View style={styles.centerContent} id="addressShow">
                    <Text style={styles.header}>
                        firstName:{this.props.address.firstName}
                    </Text>

                    <Text style={styles.header}>
                        lastName:{this.props.address.lastName}
                    </Text>

                    <Text style={styles.header}>
                        Street:{this.props.address.street}
                    </Text>

                    <Text style={styles.header}>
                        City:{this.props.address.city}
                    </Text>

                    <Text style={styles.header}>
                        State:{this.props.address.state}
                    </Text>

                    <Text style={styles.header}>
                        Zip:{this.props.address.zip}
                    </Text>

                    <Text style={styles.header}>
                        Phone:{this.props.address.phone}
                    </Text>

                    <Text style={styles.header}>
                        Websit:{this.props.address.website}
                    </Text>

                    <Text style={styles.header}>
                        Email:{this.props.address.email}
                    </Text>

                    <Text style={styles.header}>
                        Contact:{this.props.address.contact}
                    </Text>
                </View>

                <View
                    style={{
                        flexDirection: 'row',
                        justifyContent: 'center'
                    }}
                >
                    <View style={{ flex: 1 }}>
                        <Button
                            id="Backward"
                            color="darkturquoise"
                            title="<<"
                            onPress={event =>
                                this.props.setAddress(0, -1, event)
                            }
                        />
                    </View>
                    <View style={{ flex: 1 }}>
                        <Button
                            id="Backward"
                            color="darkturquoise"
                            title="<"
                            onPress={event =>
                                this.props.setAddress(-1, 0, event)
                            }
                        />
                    </View>
                    {/*<View style={{flex:1}}>*/}
                    {/*<Button*/}
                    {/*id="setAddress"*/}
                    {/*color="darkturquoise"*/}
                    {/*title="||"*/}
                    {/*onPress={event => this.props.setAddress(0,0, event)}*/}
                    {/*/>*/}
                    {/*</View>*/}
                    <View style={{ flex: 1 }}>
                        <Button
                            id="Forward"
                            color="darkturquoise"
                            title=">"
                            onPress={event =>
                                this.props.setAddress(1, 0, event)
                            }
                        />
                    </View>
                    <View style={{ flex: 1 }}>
                        <Button
                            id="Forward"
                            color="darkturquoise"
                            title=">>"
                            onPress={event =>
                                this.props.setAddress(0, 1, event)
                            }
                        />
                    </View>
                </View>
            </View>
        );
    }
}

AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        website: PropTypes.string,
        email: PropTypes.string,
        contact: PropTypes.string
    }),
    setAddress: PropTypes.func
};
