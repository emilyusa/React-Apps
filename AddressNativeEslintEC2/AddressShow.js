import React from 'react';
import { Button, Text, View } from 'react-native';
import styles from './elf-styles';
import PropTypes from 'prop-types';

export default class AddressShow extends React.Component {
    render() {
        return (
            <View style={styles.container}>
                <Text>firstName:{this.props.address.firstName}</Text>

                <Text style={styles.addressContainer}>
                    lastName:{this.props.address.lastName}
                </Text>

                <Text style={styles.addressContainer}>
                    Street:{this.props.address.street}
                </Text>

                <Text style={styles.addressContainer}>
                    City:{this.props.address.city}
                </Text>

                <Text style={styles.addressContainer}>
                    State:{this.props.address.state}
                </Text>

                <Text style={styles.addressContainer}>
                    Zip:{this.props.address.zip}
                </Text>

                <Text style={styles.addressContainer}>
                    Phone:{this.props.address.phone}
                </Text>

                <Text style={styles.addressContainer}>
                    Websit:{this.props.address.website}
                </Text>

                <Text style={styles.addressContainer}>
                    Email:{this.props.address.email}
                </Text>

                <Text style={styles.addressContainer}>
                    Contact:{this.props.address.contact}
                </Text>
                <View style={styles.buttonView}>
                    <Button
                        id="Backward"
                        title="Backward"
                        onPress={event => this.props.setAddress(-1, event)}
                    />
                    <Button
                        id="setAddress"
                        title="setAddress"
                        onPress={event => this.props.setAddress(0, event)}
                    />
                    <Button
                        id="Forward"
                        title="Forward"
                        onPress={event => this.props.setAddress(1, event)}
                    />
                </View>
            </View>
        );
    }
}

AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        website: PropTypes.string,
        email: PropTypes.string,
        contact: PropTypes.string
    }),
    setAddress: PropTypes.func
};
