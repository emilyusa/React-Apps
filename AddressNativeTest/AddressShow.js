import React from 'react';
import {Button, Text, View} from 'react-native';
import styles from "./elf-styles";


export default class AddressShow extends React.Component {
    
    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.addressContainer}>firstName:{this.props.address.firstName}
                </Text>
                
                <Text style={styles.addressContainer}>lastName:{this.props.address.lastName}
                </Text>
                
                <Text style={styles.addressContainer}>Street:{this.props.address.street}
                </Text>
                
                <Text style={styles.addressContainer}>City:{this.props.address.city}
                </Text>
                
                <Text style={styles.addressContainer}>State:{this.props.address.state}
                </Text>
                
                <Text style={styles.addressContainer}>Zip:{this.props.address.zip}
                </Text>

                <Text style={styles.addressContainer}>Phone:{this.props.address.phone}
                </Text>

                <Text style={styles.addressContainer}>Websit:{this.props.address.website}
                </Text>

                <Text style={styles.addressContainer}>Email:{this.props.address.email}
                </Text>

                <Text style={styles.addressContainer}>Contact:{this.props.address.contact}
                </Text>
                <View style={styles.buttonView}>
                <Button
                    title="setAddressLast"
                    onPress={(event) => this.props.setAddress(-1, event)}>
                </Button>
                <Button
                    title="setAddress"
                    onPress={(event) => this.props.setAddress(0, event)}>
                </Button>
                <Button
                    title="setAddressNext"
                    style={styles.buttonView}
                    onPress={(event) => this.props.setAddress(1, event)}>
                </Button>
                </View>
                              
          </View>
        );
    }
}

