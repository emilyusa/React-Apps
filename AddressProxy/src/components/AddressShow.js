import React, { Component } from 'react';
import '../css/App.css';
import RaisedButton from 'material-ui/RaisedButton';
import styles from './elf-styles';
import FontIcon from 'material-ui/FontIcon';
import { red500 } from 'material-ui/styles/colors';
import PropTypes from 'prop-types';

class AddressShow extends Component {

    render() {
        return (
            <div className="App">
                <p className="App-intro">firstName:{this.props.address.firstName}</p>
                <p className="App-intro">lastName:{this.props.address.lastName}</p>
                <p className="App-intro">Street:{this.props.address.street}</p>
                <p className="App-intro">City:{this.props.address.city}</p>
                <p className="App-intro">State:{this.props.address.state}</p>
                <p className="App-intro">Zip:{this.props.address.zip}</p>
                <p className="App-intro">Phone:{this.props.address.phone}</p>
                <p className="App-intro">Website:{this.props.address.website}</p>
                <p className="App-intro">Email:{this.props.address.email}</p>
                <p className="App-intro">Contact:{this.props.address.contact}</p>
                <RaisedButton
                    id="setAddressLast"
                    style={styles.button}
                    primary={true}
                    icon={
                        <FontIcon class="material-icons" color={red500}>
                            arrow_back_ios
                        </FontIcon>
                    }
                    onClick={(event) => this.props.setAddress(-1,event)}>

                </RaisedButton>
                <RaisedButton
                    id="setAddress"
                    label="Set Address"
                    labelPosition="before"
                    primary={true}
                    icon={
                        <FontIcon class="material-icons" color={red500}>
                            rss_feed
                        </FontIcon>
                    }
                    style={styles.button}
                    onClick={(event)=>this.props.setAddress(0,event)}>
                </RaisedButton>
                <RaisedButton
                    id="setAddressNext"
                    style={styles.button}
                    primary={true}
                    icon={
                        <FontIcon class="material-icons" color={red500}>
                            arrow_forward_ios
                        </FontIcon>
                    }
                    onClick={(event) => this.props.setAddress(1,event)}>

                </RaisedButton>
            </div>
        );
    }
}

AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street:PropTypes.string,
        city:PropTypes.string,
        state:PropTypes.string,
        zip:PropTypes.string,
        phone:PropTypes.string,
        website:PropTypes.string,
        email:PropTypes.string,
        contact:PropTypes.string
    }),
    setAddress: PropTypes.func
};

export default AddressShow;