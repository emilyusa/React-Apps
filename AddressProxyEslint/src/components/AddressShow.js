import React, { Component } from 'react';
import './App.css';
import Button from '@material-ui/core/Button';

import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    rightIcon: {
        marginLeft: theme.spacing.unit,
    }
});

class AddressShow extends Component {
    render() {
        const { classes } = this.props;
        return (
            <div className="App">
                <p className="App-intro">
                    firstName:{this.props.address.firstName}
                </p>
                <p className="App-intro">
                    lastName:{this.props.address.lastName}
                </p>
                <p className="App-intro">Street:{this.props.address.street}</p>
                <p className="App-intro">City:{this.props.address.city}</p>
                <p className="App-intro">State:{this.props.address.state}</p>
                <p className="App-intro">Zip:{this.props.address.zip}</p>
                <p className="App-intro">Phone:{this.props.address.phone}</p>
                <p className="App-intro">
                    Website:{this.props.address.website}
                </p>
                <p className="App-intro">Email:{this.props.address.email}</p>
                <p className="App-intro">
                    Contact:{this.props.address.contact}
                </p>
                <div className="App">
                    <Button
                        color='secondary'
                        variant='raised'
                        onClick={e => this.props.setAddress(-1, e)}
                    >
                        <Icon className={classes.rightIcon}>arrow_back</Icon>
                    </Button>

                    <Button
                        id='setAddress'
                        color='primary'
                        variant='raised'
                        onClick={e => this.props.setAddress(0, e)}
                    >
                        Address
                    </Button>
                    <Button
                        color='secondary'
                        variant='raised'
                        onClick={e => this.props.setAddress(1, e)}
                    >
                        <Icon className={classes.rightIcon}>arrow_forward</Icon>
                    </Button>
                </div>
            </div>
        );
    }
}

AddressShow.propTypes = {
    address: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        website: PropTypes.string,
        email: PropTypes.string,
        contact: PropTypes.string
    }),
    setAddress: PropTypes.func,
    classes: PropTypes.object.isRequired
};

export default withStyles(styles) (AddressShow);
