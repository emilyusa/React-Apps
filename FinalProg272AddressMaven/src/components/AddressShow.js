import React, { Component } from 'react';
import '../App.css';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import withStyles from '@material-ui/core/styles/withStyles';
import AddressEdit from './AddressEdit';
import PropTypes from 'prop-types';

const styles = theme => ({
    button: {
        margin: theme.spacing.unit
    },
    container: {
        flexGrow: 1,
        textAlign: 'center'
    },
    rootBar: theme.mixins.gutters({
        paddingTop: 16,
        paddingBottom: 16,
        paddingLeft: 8,
        paddingRight: 8,
        marginTop: theme.spacing.unit * 3,
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3
    })
});

class AddressShow extends Component {
    constructor(props) {
        super(props);
        this.state = {
            editOpen: false,
            edits: {
                _id: 'unknown',
                firstName: 'unknown',
                lastName: 'unknown'
            }
        };
    }

    setEdits = (name, event) => {
        var data = this.props.name;
        data[name] = event.target.value;
        this.setState({ edits: data });
    };

    addressEdit = address => {
        console.log(address);

        if (!address) {
            return this.setState({ editOpen: false });
        }

        this.setState({
            edits: address,
            editOpen: false
        });
    };

    save = () => {
        this.props.save(this.state.edits);
    };

    render() {
        const { classes } = this.props;

        const editDialog = this.state.editOpen ? (
            <AddressEdit
                address={this.props.name}
                open={this.state.editOpen}
                addressEdit={this.addressEdit}
            />
        ) : (
            <div />
        );

        return (
            <div className={classes.container}>
                <Paper className={classes.rootBar}>
                    <p>{this.props.name.firstName}</p>
                    <p>{this.props.name.lastName}</p>
                    <p>{this.props.name.street}</p>
                    <p>{this.props.name.city}</p>
                    <p>{this.props.name.state}</p>
                    <p>{this.props.name.zip}</p>
                    <p>{this.props.name.phone}</p>
                    <p>{this.props.name.website}</p>
                    <p>{this.props.name.email}</p>
                    <p>{this.props.name.contact}</p>
                    <Button
                        id="setAddress"
                        color="secondary"
                        variant="raised"
                        onClick={this.props.showAddress}
                    >
                        Show
                    </Button>

                    <Button
                        id="back"
                        color="secondary"
                        variant="raised"
                        onClick={event => this.props.setAddress(-1, event)}
                    >
                        Back
                    </Button>
                    <Button
                        color="secondary"
                        variant="raised"
                        onClick={event => this.props.setAddress(1, event)}
                    >
                        Forward
                    </Button>
                    <div>
                        <div className={classes.container}>
                            <Button
                                id="edit"
                                color="secondary"
                                variant="raised"
                                onClick={() =>
                                    this.setState({ editOpen: true })
                                }
                            >
                                Edit
                            </Button>
                            {editDialog}
                        </div>
                        <Button
                            color="secondary"
                            variant="raised"
                            onClick={this.save}
                        >
                            Save
                        </Button>
                        <Button
                            color="secondary"
                            variant="raised"
                            onClick={event =>
                                this.props.delete(this.props.name, event)
                            }
                        >
                            Delete
                        </Button>
                    </div>
                </Paper>
            </div>
        );
    }
}

AddressShow.propTypes = {
    name: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
        street: PropTypes.string,
        city: PropTypes.string,
        state: PropTypes.string,
        zip: PropTypes.string,
        phone: PropTypes.string,
        website: PropTypes.string,
        email: PropTypes.string,
        contact: PropTypes.string
    }),
    setAddress: PropTypes.func,
    showAddress: PropTypes.func,
    delete: PropTypes.func,
    save: PropTypes.func,
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(AddressShow);
