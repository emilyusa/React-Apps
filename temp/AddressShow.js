import React, { Component } from 'react';
import '../css/App.css';
import RaisedButton from 'material-ui/RaisedButton';
import styles from './elf-styles';
import FontIcon from 'material-ui/FontIcon';
import { red500 } from 'material-ui/styles/colors';
import PropTypes from 'prop-types';

class AddressShow extends Component {
    static propTypes = {
        address: PropTypes.string,
        setAddress: PropTypes.string
    };

    render() {
        return (
            <div className="App">
                <p className="App-intro">
                    firstName:{this.props.address.firstName}
                </p>
                <p className="App-intro">
                    lastName:{this.props.address.lastName}
                </p>
                <p className="App-intro">Street:{this.props.address.street}</p>
                <p className="App-intro">City:{this.props.address.city}</p>
                <p className="App-intro">
                    State:{this.props.address.somestate}
                </p>
                <p className="App-intro">Postal:{this.props.address.postal}</p>
                <RaisedButton
                    id="setAddress"
                    label="Set Address"
                    labelPosition="before"
                    primary={true}
                    icon={
                        <FontIcon class="material-icons" color={red500}>
                            rss_feed
                        </FontIcon>
                    }
                    style={styles.button}
                    onClick={this.props.setAddress}
                />
            </div>
        );
    }
}

export default AddressShow;
