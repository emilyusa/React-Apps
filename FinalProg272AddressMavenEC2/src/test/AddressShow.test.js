import React from 'react';
import AddressShow from '../components/AddressShow';
import addresses from '../address-list';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import dataManager from '../assets/FakeDataManager';

configure({ adapter: new Adapter() });

describe('AddressShow Shallow Suite', function() {
    const address = addresses[0];

    it('should take an AddressShow snapshot', () => {
        const elfTree = shallow(<AddressShow address={address}/>);
        expect(elfTree).toMatchSnapshot();
    });

    const addressProp = wrapper =>
        wrapper.find('WithStyles(AddressShow)').prop('address');

    it('renders and displays the default first name', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').firstName).toEqual('unknown');
    });

    it('renders and displays the default last name', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').lastName).toEqual('unknown');
    });

    it('renders and displays the default street', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').street).toEqual('unknown');
    });

    it('renders and displays the default city', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').city).toEqual('unknown');
    });

    it('renders and displays the default state', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').state).toEqual('unknown');
    });

    it('renders and displays the default zip', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').zip).toEqual('unknown');
    });

    it('renders and displays the default phone', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').phone).toEqual('unknown');
    });

    it('renders and displays the default website', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').website).toEqual('unknown');
    });

    it('renders and displays the default email', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').email).toEqual('unknown');
    });

    it('renders and displays the default contact', () => {
        const wrapper = shallow(<AddressShow addressList={address} />);
        expect(wrapper.prop('addressList').contact).toEqual('unknown');
    });

    const afterClickFieldTest = wrapper => {
        setImmediate(() => {
            wrapper.update();
            wrapper.instance();
            setImmediate(() => {
                wrapper.update();
            });
        });
    };

    const setAddress=()=>{
        const address=addresses[1];
        this.setProps({address:address});
    };

    it('renders and displays the first name after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).firstName).toEqual('Patty');
        });
    });

    it('renders and displays the last name after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).lastName).toEqual('Murray');
        });
    });

    it('renders and displays the street after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).street).toEqual(
                '154 Russell Senate Office Building'
            );
        });
    });

    it('renders and displays the city after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).city).toEqual('Washington DC');
        });
    });

    it('renders and displays the state after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).state).toEqual('WA');
        });
    });

    it('renders and displays the zip after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).zip).toEqual(' 20510');
        });
    });

    it('renders and displays the phone after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).phone).toEqual('202-224-2621');
        });
    });

    it('renders and displays the website after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).website).toEqual(
                'https://www.murray.senate.gov/public'
            );
        });
    });

    it('renders and displays the email after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).email).toEqual('');
        });
    });

    it('renders and displays the contact after click', () => {
        const wrapper = shallow(
            <AddressShow dataManager={dataManager} addressList={addresses} setAddress={setAddress}/>
        );
        afterClickFieldTest(wrapper, () => {
            expect(addressProp(wrapper).contact).toEqual(
                'http://www.murray.senate.gov/public/index.cfm/contactme'
            );
        });
    });
});
