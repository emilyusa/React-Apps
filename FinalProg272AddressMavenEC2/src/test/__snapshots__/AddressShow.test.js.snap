// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressShow Shallow Suite should take an AddressShow snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AddressShow)
    address={
        Object {
            "city": "unknown",
            "contact": "unknown",
            "email": "unknown",
            "firstName": "unknown",
            "lastName": "unknown",
            "phone": "unknown",
            "state": "unknown",
            "street": "unknown",
            "website": "unknown",
            "zip": "unknown",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "address": Object {
        "city": "unknown",
        "contact": "unknown",
        "email": "unknown",
        "firstName": "unknown",
        "lastName": "unknown",
        "phone": "unknown",
        "state": "unknown",
        "street": "unknown",
        "website": "unknown",
        "zip": "unknown",
      },
      "classes": Object {
        "button": "AddressShow-button-1",
        "container": "AddressShow-container-2",
        "rootBar": "AddressShow-rootBar-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "address": Object {
          "city": "unknown",
          "contact": "unknown",
          "email": "unknown",
          "firstName": "unknown",
          "lastName": "unknown",
          "phone": "unknown",
          "state": "unknown",
          "street": "unknown",
          "website": "unknown",
          "zip": "unknown",
        },
        "classes": Object {
          "button": "AddressShow-button-1",
          "container": "AddressShow-container-2",
          "rootBar": "AddressShow-rootBar-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
